flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchScheduler.java

```diff
@@ -327,6 +332,41 @@ private ParallelismAndInputInfos tryDecideParallelismAndInputInfos(
         return parallelismAndInputInfos;
     }
 
+    private List<ExecutionJobVertex> findExecutionJobVerticesUpStreamFinished() {
+        return IterableUtils.toStream(getExecutionGraph().getVerticesTopologically())
+                .filter(ExecutionJobVertex::isInitialized)
+                .filter(ejv -> !ejv.getInputs().isEmpty())
+                .filter(ejv -> tryGetConsumedResultsInfo(ejv).isPresent())
+                .collect(Collectors.toList());
+    }
+
+    private void tryEnrichInputNumBytesForExecutionVertexInputInfos() {
+        List<ExecutionJobVertex> executionJobVertices = findExecutionJobVerticesUpStreamFinished();
+
+        for (ExecutionJobVertex ejv : executionJobVertices) {
+            List<IntermediateResult> intermediateResults = ejv.getInputs();
+            for (IntermediateResult intermediateResult : intermediateResults) {
+                for (ExecutionVertex ev : ejv.getTaskVertices()) {
+                    ExecutionVertexInputInfo inputInfo =
+                            ev.getExecutionVertexInputInfo(intermediateResult.getId());
+                    if (inputInfo.getAggregatedInputNumBytes() != 0) {
+                        // current intermediate result has been calculated, we can skip it.
```

Comment:
The value may change if failover happened to the upstream tasks. So we cannot skip it.

A safer way is to compute the total input bytes when an execution vertex is scheduled. If upstream failover happens, the execution vertex will be restarted and the input bytes will be re-computed.

Reply:
fix this.